### ISSUE TITLE
[4.0.x][Bug] Crud views are broken (Laravel 6.2)

### ISSUE THREAD

**Issue by AforDesign** [Laravel-Backpack/CRUD#2143]
# Bug report

### What I did
New Laravel 6.x + Backpack v4.x installation

### What I expected to happen
Nice panel views like in the demo

### What happened
Appearently the new CRUD views have a different structure then the BASE views.
This results in missing classes, different wrappersr, and missing stylesheets, fonts and scripts.

Without any modification, the CRUD views look like:
<img width="1372" alt="Schermafbeelding 2019-10-11 om 11 41 47" src="https://user-images.githubusercontent.com/26324454/66643336-c2082b00-ec1e-11e9-9cc2-18d9fcb98c70.png">
<img width="1374" alt="Schermafbeelding 2019-10-11 om 11 42 20" src="https://user-images.githubusercontent.com/26324454/66643337-c2a0c180-ec1e-11e9-8e82-8db45d206b8f.png">

### What I've already tried to fix it
I've overridden the top_left.blade.php template with a customized version in my resourcemap to make it look more like the rest.

This file can be found in the CRUD package in '**src/resources/views/base/layouts/top_left.blade.php'**

When I run ```php artisan backpack:version``` the output is:

### PHP VERSION:
PHP 7.2.1 (cli) (built: Jan 15 2018 12:20:50) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2017 Zend Technologies

### LARAVEL VERSION:
laravel/framework                     v6.2.0             The Laravel Fram...

### BACKPACK VERSION:
backpack/backupmanager                1.4.10             Admin interface ...
backpack/base                         1.1.13             Laravel Backpack...
backpack/crud                         4.0.7              Quickly build an...
backpack/generators                   1.2.7              Generate files f...
backpack/logmanager                   2.3.27             An interface to ...
backpack/permissionmanager            v4.x-dev 65a7775   Users and permis...

**Comment by samuk190** [comment]
My crud is working in laravel 6.0. When this happens it seems a version of your base is different for your crud.

Your backpack generator is version 1, while there're version 2 for laravel 6
your backpack base is version 1, while there're version 2 for laravel 6 too.

paste this on your composer json require

    "backpack/backupmanager": "^2.0",
        "backpack/crud": "^4.0",
        "backpack/logmanager": "^3.0",
        "backpack/pagemanager": "^2.0",
        "backpack/permissionmanager": "^5.0",
        "backpack/settings": "^3.0",
        "barryvdh/laravel-elfinder": "^0.4.3",
        "fideloper/proxy": "^4.0",
        "infyomlabs/adminlte-templates": "6.0.x-dev",
        "infyomlabs/generator-builder": "dev-master",
        "infyomlabs/laravel-generator": "6.0.x-dev",
        "infyomlabs/swagger-generator": "dev-master",
        "jlapp/swaggervel": "dev-master",
        "laravel/framework": "^6.0",
        "laravel/tinker": "^1.0",
        "laravel/ui": "^1.0",
        "laravelcollective/html": "^6.0",
        "lavary/laravel-menu": "^1.7",
        "yajra/laravel-datatables-buttons": "^4.0",
        "yajra/laravel-datatables-oracle": "^9.6"

and in your require-dev
  "require-dev": {
        "backpack/generators": "^2.0",
        "facade/ignition": "^1.4",
        "fzaninotto/faker": "^1.4",
        "mockery/mockery": "^1.0",
        "nunomaduro/collision": "^3.0",
        "phpunit/phpunit": "^8.0",
        "sven/artisan-view": "^3.3"

run composer update, and publish all files again. it will fix everything

**Comment by tabacitu** [comment]
Indeed, Backpack/CRUD 4.x does NOT need Backpack/Base. I see you have both installed, so that must be the problem - both Base and CRUD define layouts etc, there's a conflict and your app seems to be picking up the CSS and JS from Base when it should be picking up those from CRUD.

Remove Backpack/Base from your ```composer.json```, then run ```composer update``` and ```php artisan view:clear```, that should fix it for you. 

@AforDesign can you please tell me which guide you used to install Backpack? The link please? There must be a problem there where you're instructed to install Base, but I can't find that anywhere. Thanks!

**Comment by samuk190** [comment]
> Indeed, Backpack/CRUD 4.x does NOT need Backpack/Base. I see you have both installed, so that must be the problem - both Base and CRUD define layouts etc, there's a conflict and your app seems to be picking up the CSS and JS from Base when it should be picking up those from CRUD.
> 
> Remove Backpack/Base from your `composer.json`, then run `composer update` and `php artisan view:clear`, that should fix it for you.
> 
> @AforDesign can you please tell me which guide you used to install Backpack? The link please? There must be a problem there where you're instructed to install Base, but I can't find that anywhere. Thanks!

thanks for your reply, I didnt know that, maybe thats the reason I have a field bug in my project

**Comment by AforDesign** [comment]
Thanks for the quick replies.
My holidays just started, so I haven't had much time to look into it untill now.

I've removed the 'Base' entry from composer.json, cleared the vendor directory and ran the composer update. Cleared all caches.
It did not help though.

@tabacitu I've followed the instructions on:
https://backpackforlaravel.com/docs/4.0/installation

Also I've installed the following official addons:
PermissionManager
Settings 
PageManager
LogManager 
BackupManager 


If I go to my Category admin (my most basic model with crud atm), the following views are loaded (copy pasted from debugbar):

crud::list (vendor/backpack/crud/src/resources/views/crud/list.blade.php)2blade
crud::inc.button_stack (vendor/backpack/crud/src/resources/views/crud/inc/button_stack.blade.php)9blade
crud::buttons.create (vendor/backpack/crud/src/resources/views/crud/buttons/create.blade.php)12blade
crud::buttons.reorder (vendor/backpack/crud/src/resources/views/crud/buttons/reorder.blade.php)12blade
crud::inc.datatables_logic (vendor/backpack/crud/src/resources/views/crud/inc/datatables_logic.blade.php)11blade
crud::inc.export_buttons (vendor/backpack/crud/src/resources/views/crud/inc/export_buttons.blade.php)11blade
crud::inc.details_row_logic (vendor/backpack/crud/src/resources/views/crud/inc/details_row_logic.blade.php)11blade
backpack::layouts.top_left (vendor/backpack/crud/src/resources/views/base/layouts/top_left.blade.php)11blade
backpack::inc.head (vendor/backpack/base/src/resources/views/inc/head.blade.php)11blade
backpack::inc.main_header (vendor/backpack/base/src/resources/views/inc/main_header.blade.php)11blade
backpack::inc.menu (vendor/backpack/base/src/resources/views/inc/menu.blade.php)11blade
backpack::inc.topbar_left_content (vendor/backpack/base/src/resources/views/inc/topbar_left_content.blade.php)11blade
backpack::inc.topbar_right_content (vendor/backpack/base/src/resources/views/inc/topbar_right_content.blade.php)11blade
backpack::inc.sidebar (vendor/backpack/base/src/resources/views/inc/sidebar.blade.php)11blade
backpack::inc.sidebar_user_panel (vendor/backpack/base/src/resources/views/inc/sidebar_user_panel.blade.php)11blade
backpack::inc.sidebar_content (resources/views/vendor/backpack/base/inc/sidebar_content.blade.php)11blade
backpack::inc.breadcrumbs (vendor/backpack/crud/src/resources/views/base/inc/breadcrumbs.blade.php)11blade
backpack::inc.footer (vendor/backpack/base/src/resources/views/inc/footer.blade.php)11blade
backpack::inc.scripts (vendor/backpack/base/src/resources/views/inc/scripts.blade.php)11blade

So, there are still a lot of base views loaded.

My Controller:
`<?php

namespace App\Http\Controllers\Admin;

use App\Http\Requests\CategoryRequest;
use Backpack\CRUD\app\Http\Controllers\CrudController;
use Backpack\CRUD\app\Library\CrudPanel\CrudPanelFacade as CRUD;

/**
 * Class CategoryCrudController
 * @package App\Http\Controllers\Admin
 * @property-read CrudPanel $crud
 */
class CategoryCrudController extends CrudController
{
    // override columnnames for reorder
    use \Backpack\CRUD\app\Http\Controllers\Operations\ListOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\CreateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\UpdateOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\DeleteOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\ShowOperation;
    use \Backpack\CRUD\app\Http\Controllers\Operations\ReorderOperation;

    public function setup()
    {
        $this->crud->setModel('App\Models\Category');
        $this->crud->setRoute(config('backpack.base.route_prefix') . '/category');
        $this->crud->setEntityNameStrings('category', 'categories');
    }

    protected function setupListOperation()
    {
        // TODO: remove setFromDb() and manually define Columns, maybe Filters
//        $this->crud->setFromDb();
        $this->crud->setColumns(['name']);
    }

    protected function setupCreateOperation()
    {
        $this->crud->setValidation(CategoryRequest::class);

        // TODO: remove setFromDb() and manually define Fields
        $this->crud->setFromDb();
    }

    protected function setupUpdateOperation()
    {
        $this->setupCreateOperation();
    }

    protected function setupReorderOperation()
    {
        // define which model attribute will be shown on draggable elements
        $this->crud->set('reorder.label', 'name');
        // define how deep the admin is allowed to nest the items
        // for infinite levels, set it to 0
        $this->crud->set('reorder.max_level', 0);
    }
}
`
My model:

`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use Illuminate\Database\Eloquent\Model;
use Spatie\Sluggable\HasSlug;
use Spatie\Sluggable\SlugOptions;

class Category extends Model
{
    /*
    |--------------------------------------------------------------------------
    | TRAITS
    |--------------------------------------------------------------------------
    */
    use CrudTrait;
    use HasSlug;

    /*
    |--------------------------------------------------------------------------
    | GLOBAL VARIABLES
    |--------------------------------------------------------------------------
    */
    protected $table = 'categories';
    // protected $primaryKey = 'id';
    // public $timestamps = false;
    protected $guarded = ['id'];
     protected $fillable = [
         'ref',
         'name',
         'slug'
     ];
    // protected $hidden = [];
    // protected $dates = [];

    /*
    |--------------------------------------------------------------------------
    | FUNCTIONS
    |--------------------------------------------------------------------------
    */
    /**
     * spatie/laravel-sluggable
     * Get the options for generating the slug.
     */
    public function getSlugOptions() : SlugOptions
    {
        return SlugOptions::create()
            ->generateSlugsFrom('name')
            ->saveSlugsTo('slug');
    }

    /**
     * spatie/laravel-sluggable
     * Get the route key for the model.
     *
     * @return string
     */
    public function getRouteKeyName()
    {
        return 'slug';
    }

    /*
    |--------------------------------------------------------------------------
    | RELATIONS
    |--------------------------------------------------------------------------
    */

    /*
    |--------------------------------------------------------------------------
    | SCOPES
    |--------------------------------------------------------------------------
    */

    /*
    |--------------------------------------------------------------------------
    | ACCESSORS
    |--------------------------------------------------------------------------
    */

    /*
    |--------------------------------------------------------------------------
    | MUTATORS
    |--------------------------------------------------------------------------
    */
}
`


Next weekend I'll be able to  start a completely fresh install and watch if the problem occurs again and at which point.

**Comment by tabacitu** [comment]
Hmm that's weird @AforDesign . I have a number of projects upgraded from Backpack 3.x to 4.0 and they work fine. So there shouldn't be a need for you to start from scratch. Something must be off here.

Your controller, model, look ok. What does your ```composer.json``` look like?

**Comment by ApexMuse** [comment]
@AforDesign - I had this same issue while going through the upgrade process.  Make sure to focus on [this section](https://backpackforlaravel.com/docs/4.0/upgrade-guide#views) in the upgrade guide, particularly step #16.

**Comment by AforDesign** [comment]
@ApexMuse Yesss! Thank you for pointing out these instructions.

Although I didn't update from v3.x, but started with v4.0 for my project, it still helped me.
The update instructions contained the step(s) that fixed my issues.

It weren't step 15 & 16 that made a difference, so I decided to check each step of this update instruction. 

2 things caught my attention:

1. In my setup, my crud config file looked very different, so I replaced it with the one linked in the instruction manual.
2. In my require-dev section, the backpack/generator was set to v1.2.4. (but it was v2.0 when I started this topic. Don't know what changed it)


So, thanks alot everyone!
I wish I could pinpoint what exactly caused the problem, but it looks like something went wrong in the original installation, without error notification.
My guess would be that the problem layed within the crud-config file. My version didn't have the correct structure of the operation settings.
